coreblocks/transactions/core.py

```diff
@@ -819,7 +819,18 @@ def decorator(func: Callable[..., Optional[RecordDict]]):
         with method.body(m, ready=ready, out=out) as arg:
             ret_out = method_def_helper(method, func, arg, **arg.fields)
 
-        if ret_out is not None:
+        if ret_out is None:
+            ret_out = Record([])
```

Comment:
```suggestion
            return
```
am I reading correctly?

Reply:
Nope. If the method definition didn't return anything, we still want to run `assign(out, ret_out)` so that it can throw an exception if `out` isn't an empty layout (the method was supposed to return something but it didn't), but `ret_out` can't be `None` so I use type-correct equivalent - empty layout.